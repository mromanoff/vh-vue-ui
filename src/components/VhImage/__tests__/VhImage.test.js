import { shallowMount } from '@vue/test-utils';
import VhImage from '../VhImage.vue';

describe('VhImage.vue', () => {
  let wrapper;

  it('renders default VhImage', () => {
    wrapper = shallowMount(VhImage, {
      propsData: {
        src: 'http:testdomain.vehiclehistory.image',
        alt: 'dummy url image',
      },
    });
    expect(wrapper.props().src).toBe('http:testdomain.vehiclehistory.image');
    expect(wrapper.props().alt).toBe('dummy url image');
    expect(wrapper.props().padded).toBe(false);
    expect(wrapper.props().width).toBe(null);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhImage', () => {
    wrapper = shallowMount(VhImage, {
      propsData: {
        src: 'http:testdomain.vehiclehistory.image',
        alt: 'dummy url image',
        padded: true,
        width: 100,
      },
    });
    expect(wrapper.props().src).toBe('http:testdomain.vehiclehistory.image');
    expect(wrapper.props().alt).toBe('dummy url image');
    expect(wrapper.props().padded).toBe(true);
    expect(wrapper.props().width).toBe(100);
    expect(wrapper).toMatchSnapshot();
  });
});
