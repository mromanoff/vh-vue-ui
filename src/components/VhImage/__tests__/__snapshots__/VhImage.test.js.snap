// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhImage.vue renders default VhImage 1`] = `
<div class="VhImage">
  <div class="VhImage-imageContainer"><img src="http:testdomain.vehiclehistory.image" alt="dummy url image" class="VhImage-image"></div>
</div>
`;

exports[`VhImage.vue renders modified VhImage 1`] = `
<div class="VhImage VhImage--padded VhImage--setWidth">
  <div class="VhImage-imageContainer" style="width: 100px; max-width: 100px;"><img src="http:testdomain.vehiclehistory.image" alt="dummy url image" class="VhImage-image"></div>
</div>
`;
